learning_journal/learning_journal/views.py

```diff
@@ -0,0 +1,95 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+import transaction
+import datetime
+from jinja2 import Markup
+import markdown
+from sqlalchemy import update
+from .forms import EntryCreateForm, EntryUpdateForm
+from pyramid.httpexceptions import HTTPFound
+
+
+from sqlalchemy.exc import DBAPIError, ResourceClosedError, IntegrityError
+
+from .models import (
+    DBSession,
+    Entry,
+)
+
+
+# @view_config(route_name='list', renderer='string')
+@view_config(route_name='list', renderer='templates/pretty.jinja2')
+def list_view(request):
+    display = DBSession().query(Entry.metadata.tables['entries']).all()[::-1][:10]
+    return {"content": display, "header": "entries.order_by(latest)[:10]"}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request):
+    id_ = request.matchdict.get('entry_id')
+    display = DBSession().query(Entry.metadata.tables['entries']).filter_by(id=id_).one()
+    md = markdown.Markdown(safe_mode='replace', html_replacement_text='--RAW HTML NOT ALLOWED--')
+    message = md.convert(display[2])
+
+    return {"message": message, "header": display[0], "title": display[1], "time": display[3]}
+
+
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
+def edit_view(request):
+    id_ = request.matchdict.get('entry_id')
+    entry = DBSession().query(Entry).get(id_)
+    id_ = entry.id
+    title = entry.title
+    text = entry.text
+    time = entry.created
+
+    form = EntryUpdateForm(request.POST, entry)
+    session = DBSession()
+    if request.method == 'POST' and form.validate():
+        form.populate_obj(entry)
+        session.add(entry)
+        session.flush()
+        entry_id = entry.id
+        transaction.commit()
+        return HTTPFound(location='/view/{}'.format(entry_id))
+    return {'message': text, 'id': id_, 'title': title, 'time': time}
```

Comment:
don't pull the attributes off of your model and pass them back.  Pass the model instance itself.